doc/man3/SSL_CTX_set_tlsext_servername_callback.pod

```diff
@@ -39,16 +41,26 @@ SSL_get_servername_type() returns the servername type or -1 if no servername
 is present. Currently the only supported type (defined in RFC3546) is
 B<TLSEXT_NAMETYPE_host_name>.
 
+SSL_set_tlsext_host_name() sets the server name indication extension in
+ClientHello with value B<name>. The type of server name indication extension
+is set to B<TLSEXT_NAMETYPE_host_name> (defined in RFC3546).
```

Comment:
Pedantically, "with value B<name>" is not correct, for the server name indication extension (because of the name type silliness that you go on to mention), but I think any more-technically-correct wording would also be more confusing, so I would leave this as-is in our current version.